src/js/Analytics.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+var Analytics = function (trackingId, userId, appName, appVersion, buildType, optOut, debugMode) {
+    this._trackingId = trackingId;
+
+    this.setOptOut(optOut);
+
+    this._analytics = googleAnalytics;
+
+    this._analytics.initialize(this._trackingId, {
+        storage: 'none',
+        clientId: userId,
+        debug: !!debugMode
+    });
+
+    // Make it work for the Chrome App:
+    this._analytics.set('forceSSL', true);
+    this._analytics.set('transport', 'xhr');
+
+    // Make it work for NW.js:
+    this._analytics.set('checkProtocolTask', null);
+
+    this._analytics.set('appName', appName);
+    this._analytics.set('appVersion', debugMode ? appVersion + '-debug' : appVersion);
+
+    this.EVENT_CATEGORIES = {
+        APPLICATION: 'Application',
+        FLIGHT_CONTROLLER: 'FlightController',
+        FIRMWARE: 'Firmware',
+    };
+
+    this.DATA = {
+        BOARD_TYPE: 'boardType',
+        FIRMWARE_TYPE: 'firmwareType',
+        FIRMWARE_VERSION: 'firmwareVersion',
+        API_VERSION: 'apiVersion',
+        MCU_ID: 'mcuId',
+        FIRMWARE_NAME: 'firmwareName',
+        FIRMWARE_CHECKSUM: 'firmwareChecksum',
+        FIRMWARE_SOURCE: 'firmwareSource',
+        FIRMWARE_CHANNEL: 'firmwareChannel',
+        FIRMWARE_ERASE_ALL: 'firmwareEraseAll',
+    };
+
+    this.DIMENSIONS = {
+        CONFIGURATOR_BUILD_TYPE: 1,
+        BOARD_TYPE: 2,
+        FIRMWARE_TYPE: 3,
+        FIRMWARE_VERSION: 4,
+        API_VERSION: 5,
+        FIRMWARE_NAME: 6,
+        FIRMWARE_SOURCE: 7,
+        FIRMWARE_ERASE_ALL: 8,
+        CONFIGURATOR_EXPERT_MODE: 9,
+        FIRMWARE_CHANNEL: 10,
+    };
+
+    this.setDimension(this.DIMENSIONS.CONFIGURATOR_BUILD_TYPE, buildType);
+
+    this.resetFlightControllerData();
+    this.resetFirmwareData();
+};
+
+Analytics.prototype.setDimension = function (dimension, value) {
+    var dimensionName = 'dimension' + dimension;
+    this._analytics.custom(dimensionName, value);
+}
+
+Analytics.prototype.sendEvent = function (category, action, options) {
+//    options.eventLabel = options.eventLabel || this._flightControllerData[this.DATA.MCU_ID];
```

Comment:
Why is this line commented? Best add a comment explaining it or remove the line.

Reply:
Hell yeah, looks like our best practices work! This a leftover from debugging, and should have been removed. Will change, thanks for the review.